src/lib/support/ZclString.h

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/util/af-types.h>
+
+namespace chip {
+
+/**
+ * @brief Create ZCL string from char
+ */
+template <size_t BufferLength>
+uint8_t * MakeZclCharString(uint8_t (&zclString)[BufferLength], char * cString)
+{
+    static_assert(BufferLength <= 254, "Too long string to fit in ZCL_CHAR_STRING type");
+    zclString[0] = static_cast<uint8_t>(strlen(cString));
```

Comment:
So if cString is longer than 255 chars we will grab some small prefix off the front?  In particular if it's 256 chars we will end up with a zero-length string?

What should probably happen here for now, to be consistent with other places that do this stuff, is that we should use `std::min(254, strlen(cString))` as the length.

Reply:
Added tests.

Reply:
I've covered 64 char length, 254 char length and 255 char length.
In case now char is 255 length there won't be zero-length char at the end, so I set ZCL String length to 0.